api/reporting/schemas/reporting-codegen.rb

```diff
@@ -26,64 +27,187 @@ def to_output_path(dir_name, table_name, suffix)
+    'datetime' => {
+        :bigquery => 'TIMESTAMP',
+        :java => 'Timestamp'
+    },
+    'bigint' => {
+        :bigquery => 'INT64',
+        :java => 'Long'
+    },
+    'smallint' => {
+        :bigquery => 'INT64',
+        :java => 'Short'
+    },
+    'longtext' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    },
+    'int' => {
+        :bigquery => 'INT64',
+        :java => 'Integer'
+    },
+    'tinyint' => {
+        :bigquery => 'INT64',
+        :java => 'Short'
+    },
+    'bit' => {
+        :bigquery => 'BOOLEAN',
+        :java => 'Boolean'
+    },
+    'double' => {
+        :bigquery =>  'FLOAT64',
+        :java => 'Double'
+    },
+    'text' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    },
+    'mediumblob' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    }
+}.freeze
+
+# Enumerated types should be the enum in Java and the DTO, but a STRING in BQ.
+# We could either try to remap to the enum type when writing to BQ, but we still
+# have to handle getting the value into the projection, and the only way I've been
+# able to do that is to use the exact same type.
+ENUM_TYPES = {
+     'workspace' => {
+        'billing_status' => {
+            :java => 'BillingStatus',
+            :constant => 'BillingStatus.ACTIVE'
+        },
+        'billing_account_type' => {
+            :java => 'BillingAccountType',
+            :constant => 'BillingAccountType.FREE_TIER'
+        }
+     },
+     'user' => {
+         'data_access_level' => {
+             :java => 'DataAccessLevel',
+             :constant => 'DataAccessLevel.REGISTERED'
+         },
+         'email_verification_status' => {
+             :java => 'EmailVerificationStatus',
+             :constant => 'EmailVerificationStatus.SUBSCRIBED'
+         }
+     }
+}.freeze
+
+# strip size/kind
+def simple_mysql_type(mysql_type)
   type_pattern = Regexp.new("(?<type>\\w+)(\\(\\d+\\))?")
   match_data = mysql_type.match(type_pattern)
-  result = MYSQL_TO_BIGQUERY_TYPE[match_data[:type]]
+  result = match_data[:type]
   raise "MySQL type #{mysql_type} not recognized." if result.nil?
   result
 end
 
-excluded_fields = File.exist?(inputs[:exclude_columns]) \
-  ? File.readlines(inputs[:exclude_columns]) \
+excluded_fields = File.exist?(INPUTS[:excluded_columns]) \
+  ? File.readlines(INPUTS[:excluded_columns]) \
       .map{ |c| c.strip } \
   : []
 
 def include_field?(excluded_fields, field)
   !excluded_fields.include?(field)
 end
 
+ENTITY_MODIFIED_COLUMNS = {
+    'workspace' => {
+        'cdr_version_id' => 'cdrVersion.cdrVersionId AS cdrVersionId',
+        'creator_id' => 'creator.userId AS creatorId',
+        'needs_rp_review_prompt' => 'needsResearchPurposeReviewPrompt AS needsRpReviewPrompt'
+    }
+}
+
 ## BigQuery schema
-describe_rows = CSV.new(File.read(inputs[:describe_csv])).sort_by { |row| row[0]  }
+describe_rows = CSV.new(File.read(INPUTS[:describe_csv])).sort_by { |row| row[0]  }
```

Comment:
Where is this input format documented?

Reply:
It's just this: https://dev.mysql.com/doc/refman/8.0/en/explain.html

Reply:
This was less a question of personal curiosity and more: "how is the next user or reader of this code going to know what the expected input format is? Please add a comment or documentation"